### DISCUSSION TITLE
How to add an external JavaScript file to the head section (namely Google AdSense)?

### DISCUSSION THREAD

**Discussion by jepster** [nuxt/nuxt#19273] [Questions] [answered]
Hi,

I would like to add an external JavaScript file to the head section of my site. Namely it's the Google AdSense JavaScript file.

Google wants me to add the following code snippet into the head area of my site:

```
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-123456789" crossorigin="anonymous"></script>
```

Currently there are only instructions for Nuxt 2 (https://pipinghot.dev/tutorial/how-to-add-google-adsense-to-nuxt/) and there is a module for a Nuxt 2 (https://www.npmjs.com/package/@nuxtjs/google-adsense). But for Nuxt 3 I cannot find any hints via Google. 

Please describe how I can add the code snippet described above.

**Comment by TheAlexLichter** [comment]
The module is up to date for Nuxt 3 too (I'm the current maintainer) ☺️

**Reply by jepster** [reply]
Well, does not work. AdSense is reporting that their code cannot be recognized on my site. Tried multiple times for making sure that I didn't overlooked something.

**Reply by TheAlexLichter** [reply]
Weird. It works fine on https://github.com/albionstatus/albionstatus-website (code + website open source). Feel free to give it a spin 👍

**Comment by AloisSeckar** [comment, marked as answer]
How about [useHead](https://nuxt.com/docs/api/composables/use-head) composable? It can be used to include scripts among other things:

```
useHead({
  script: [
    {
      async: true,
      src: 'https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-123456789',
      crossorigin: 'anonymous',
    },
  ],
});
```

Demo here, I think its included (at least my AdBlock says so :D)  https://stackblitz.com/edit/github-dszdjm?file=app.vue

**Reply by TheAlexLichter** [reply]
Yes, that'd be the way to go for *any* script though for adsense, the module has a few more advantages (e.g. reloading ads when navigating)

**Reply by jepster** [reply]
Thank you @AloisSeckar! That worked immediately. Google AdSense is recognizing their snippet now on my site. So happy. I was searching and trying the Google AdSense module for hours. Finally it works. :)

**Comment by videmann** [comment]
You also can define it in nuxt.config.ts, app.head section. just add key: script(s): object[] and insert your scripts
